src/core/tsi/ssl_transport_security.cc

```diff
@@ -2161,7 +2161,7 @@ static void tsi_ssl_client_handshaker_factory_destroy(
   self->session_cache.reset();
```

Comment:
Don't actually need these three lines anymore, since the dtors will now be called.